### DISCUSSION TITLE
http and https return same result

### DISCUSSION THREAD

**Discussion by stanislavprokopov** [umputun/reproxy#107] [Q&A] [answered]
I have Reproxy latest version with ssl.auto.

Can it be configured so that when client enters http://example.com he is not auto redirected to https://example.com with Temporary Redirect? I need to serve both http and https urls with same result.

Looking at the code this is the place that causes this redirect to happen:
https://github.com/umputun/reproxy/blob/211fe15b8fff3e3087aa67233c8ed780b9ad8f0e/app/proxy/proxy.go#L146
https://github.com/umputun/reproxy/blob/211fe15b8fff3e3087aa67233c8ed780b9ad8f0e/app/proxy/ssl.go#L51
https://github.com/umputun/reproxy/blob/211fe15b8fff3e3087aa67233c8ed780b9ad8f0e/app/proxy/ssl.go#L56

**Comment by umputun** [comment, marked as answer]
With ssl mode, reproxy redirects http to https unless this is LE http challenge request. This is done intentionally as the majority of use cases I can think of expect such a behavior. There is no flag/parmeter to alter it.

If you can describe your use case, i.e. why you want to serve your content over HTTP as well, and if there is a good reason to do it I may consider adding a flag disabling this `redirectHandler`

**Reply by stanislavprokopov** [reply]
The reason is that a third-party app that allows to enter an API url for some strange reason ignores the fact that it is entered with https in front and keeps making the requests to http domain. As there is no way to fix this app, the only option i am left with is to support both http and https.

**Reply by umputun** [reply]
well, this is really exotic use case. I don't think supporting it directly on the reproxy side is a good idea. Maybe you can run the second reproxy on some other port as http-only as a workaround  and point your client to this one

**Reply by stanislavprokopov** [reply]
Well this client also removes ports, so thats not an option. Looks like i will need to find another way to solve this for me or maybe hack my own build of reproxy so it works the way i need it to.

Thanks for support, this QA can be closed now.

**Reply by fess932** [reply]
android 4.4 not wokring with ssl 1.2
